src/ERC721Drop.sol

```diff
@@ -1267,6 +1275,36 @@ contract ERC721Drop is
         emit MintFeePayout(zoraFee, ZORA_MINT_FEE_RECIPIENT, success);
     }
 
+    function _requireCanMintQuantity(uint256 quantity) internal view {
+        if (quantity + _totalMinted() > config.editionSize) {
+            revert Mint_SoldOut();
+        }
+    }
+
+    function _mintSupplyRoyalty(uint256 mintQuantity) internal {
+        if (royaltyMintSchedule == 0) {
+            return;
+        }
+
+        address royaltyRecipient = config.fundsRecipient;            
+        if (royaltyRecipient == address(0)) {
+            return;
+        }
+
+        uint256 totalRoyaltyMints = (mintQuantity + (_totalMinted() % royaltyMintSchedule)) / (royaltyMintSchedule - 1);
+        totalRoyaltyMints = MathUpgradeable.min(totalRoyaltyMints, config.editionSize - (mintQuantity + _totalMinted()));
```

Comment:
Should we have comments for this?

Reply:
its @tbtstl 's whiteboard magic, would be great if you had any notes on it tyson.